### ISSUE TITLE
testStaticOptimization and testCMCGait10dof18musc fail on fresh build

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-core#784]
(Win7, VS2015 32-bit, Release mode) I followed the instructions in README.md including the instructions for superbuild (not [this one](http://www.superbuild.co.nz/)), with the following exception: in step 9, running `cmake --build . --config Release` returned `Error: could not find CMAKE_MAKE_PROGRAM in Cache` so I built in Visual Studio instead. All but two tests pass; output follows.

_testStaticOptimization_
```
...
Loading coordinates from file walk_subject01_Kinematics_q.sto.
Storage: file=walk_subject01_Kinematics_q.sto (nr=2537 nc=11)
Found 2536 state vectors with time stamps ranging from 0 to 2.499.
Executing the analyses from 0.5 to 1.3...
SimTK Exception thrown at InteriorPointOptimizer.cpp:264:
  Optimizer failed: Ipopt: Maximum iterations exceeded (status -1)
OPTIMIZATION FAILED...
...
```

_testCMCGait10dof18musc_
```
...
================================================================
Computing initial values for muscles states (activation, length)
Start time = Sat Feb 27 00:41:56 2016

================================================================
CMC.computeControls:  t = 0.6
CMC::computeControls WARNING- small force range for iliopsoas_r (0 to 0)

CMC::computeControls WARNING- small force range for rect_fem_r (0 to 0)

CMC::computeControls WARNING- small force range for gastroc_r (0 to 0)

CMC::computeControls WARNING- small force range for soleus_r (0 to 0)

CMC::computeControls WARNING- small force range for iliopsoas_l (0 to 0)

CMC::computeControls WARNING- small force range for rect_fem_l (0 to 0)

CMC::computeControls WARNING- small force range for gastroc_l (0 to 0)

CMC::computeControls WARNING- small force range for soleus_l (0 to 0)

SimTK Exception thrown at InteriorPointOptimizer.cpp:264:
  Optimizer failed: Ipopt: Infeasible problem detected (status 2)
OPTIMIZATION FAILED...
...
```

**Comment by chrisdembia** [comment]
> (not this one)

haha

Hmm...you built 64-bit libraries or 32-bit libraries (I'm not sure if "VS2015 32-bit" refers to the fact that VS itself is 32-bit)? We only test 64-bit libraries on Appveyor, so maybe it's because you might be building as 32-bit?

**Comment by tkuchida** [comment]
VS is 32-bit and I built 32-bit libraries. Is this configuration no longer supported?

testStaticOptimization failed with `Ipopt: Maximum iterations exceeded (status -1)`. This failure could make sense if the 32- and 64-bit versions make different amounts of progress per iteration---is this so? Note, however, that all the [`testArm26()`](https://github.com/opensim-org/opensim-core/blob/2e0e9e8690a02f034f4eea8f38eb79faa47cd17c/Applications/Analyze/test/testStaticOptimization.cpp#L97) tests in testStaticOptimization pass; the failure occurs in [`testModelWithPassiveForces()`](https://github.com/opensim-org/opensim-core/blob/2e0e9e8690a02f034f4eea8f38eb79faa47cd17c/Applications/Analyze/test/testStaticOptimization.cpp#L180).

**Comment by chrisdembia** [comment]
> Is this configuration no longer supported?

I think we intend to support it; I just don't know how often it is tested.

> is this so?

no clue

**Comment by tkuchida** [comment]
Still stuck. Potentially relevant:
- `testModelWithPassiveForces()` was added to testStaticOptimization in #546, merged 5 months ago. Has anyone tested in 32-bit since then (i.e., has the test ever been run in 32-bit)?
- Toggling the `use_muscle_physiology` property from `true` to `false` in [the Analyze setup file](https://github.com/opensim-org/opensim-core/blob/master/Applications/Analyze/test/staticoptimization_spring_Setup.xml#L47) called [here](https://github.com/opensim-org/opensim-core/blob/2e0e9e8690a02f034f4eea8f38eb79faa47cd17c/Applications/Analyze/test/testStaticOptimization.cpp#L181) (which seems innocuous enough) results in the infamous DLASD4 "parameter number -1 had an illegal value" error. My understanding from [this Simbody issue](https://github.com/simbody/simbody/issues/177) is that this bug wasn't patched for 32-bit, but that the problem affected only the way errors are handled so is irrelephant here. :elephant:

**Comment by aymanhab** [comment]
@tkuchida Test was added as part of release 3.3 and it was passing for 32 bit environment with VisualStudio2015 using the 3.2 codebase. Not sure if it was ever run in 32 bit environment in master.  

There were two additions to the test (PassiveForces and DLASD test), are they both failing?

**Comment by chrisdembia** [comment]
@tkuchida it's not clear to me; did you try building in 64-bit and the tests still failed?



> Toggling the use_muscle_physiology property from true to false in the Analyze setup file called here (which seems innocuous enough) results in the infamous DLASD4 "parameter number -1 had an illegal value" error

Oh noo!!!

> irrelephant

:)

**Comment by aymanhab** [comment]
DLASD4 error was claimed to be 64 bit lapack specific and was never encountered in 32 bit lapack. Since we don't build or distribute lapack ourselves, is it possible that the wrong one (64 bit) is being used in the 32 bit environment?

**Comment by chrisdembia** [comment]
We do distribute lapack (through Simbody).

**Comment by sherm1** [comment]
64 bit lapack wouldn't link with a 32 bit executable. So this is likely the 32 bit one, without the error handling fix that we put in the 64 bit one. The bug was definitely present in 32 bit, we just hadn't seen it get triggered.

**Comment by tkuchida** [comment]
All tests now pass locally (Win7, VS2015, 32-bit) on a clean build of master. The test failures may have been caused by the same bug that caused the hackathon example to fail (which was fixed by @aseth1 in #894). I have added a 32-bit build to AppVeyor (#921), which also currently passes, so I think this issue can be closed cleanly. :shower:
